account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,1190 @@
+            return parsed_inv
+        prec_ac = self.env["decimal.precision"].precision_get("Account")
+        prec_pp = self.env["decimal.precision"].precision_get("Product Price")
+        prec_uom = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        if "amount_tax" in parsed_inv and "amount_untaxed" not in parsed_inv:
+            parsed_inv["amount_untaxed"] = (
+                parsed_inv["amount_total"] - parsed_inv["amount_tax"]
+            )
+        elif "amount_untaxed" not in parsed_inv and "amount_tax" not in parsed_inv:
+            # For invoices that never have taxes
+            parsed_inv["amount_untaxed"] = parsed_inv["amount_total"]
+        # Support the 2 refund methods; if method a) is used, we convert to
+        # method b)
+        if not parsed_inv.get("type"):
+            parsed_inv["type"] = "in_invoice"  # default value
+        if (
+            parsed_inv["type"] == "in_invoice"
+            and float_compare(parsed_inv["amount_total"], 0, precision_digits=prec_ac)
+            < 0
+        ):
+            parsed_inv["type"] = "in_refund"
+            for entry in ["amount_untaxed", "amount_total"]:
+                parsed_inv[entry] *= -1
+            for line in parsed_inv.get("lines", []):
+                line["qty"] *= -1
+                if "price_subtotal" in line:
+                    line["price_subtotal"] *= -1
+        # Handle the case where we import an invoice with VAT in a company that
+        # cannot deduct VAT
+        if self.company_cannot_refund_vat():
+            parsed_inv["amount_tax"] = 0
+            parsed_inv["amount_untaxed"] = parsed_inv["amount_total"]
+            for line in parsed_inv.get("lines", []):
+                if line.get("taxes"):
+                    if len(line["taxes"]) > 1:
+                        raise UserError(
+                            _(
+                                "You are importing an invoice in a company that "
+                                "cannot deduct VAT and the imported invoice has "
+                                "several VAT taxes on the same line (%s). We do "
+                                "not support this scenario for the moment."
+                            )
+                            % line.get("name")
+                        )
+                    vat_rate = line["taxes"][0].get("amount")
+                    if not float_is_zero(vat_rate, precision_digits=2):
+                        line["price_unit"] = line["price_unit"] * (1 + vat_rate / 100.0)
+                        line.pop("price_subtotal")
+                        line["taxes"] = []
+        # Rounding work
+        for entry in ["amount_untaxed", "amount_total"]:
+            parsed_inv[entry] = float_round(parsed_inv[entry], precision_digits=prec_ac)
+        for line in parsed_inv.get("lines", []):
+            line["qty"] = float_round(line["qty"], precision_digits=prec_uom)
```

Comment:
There is an exception possibility here:

Invoice line qty = 0,5 / price = 100
Product Unit of Measure = 0

When importing the quantity will be set to 1, but the price is not changed accordingly. Result is an unclear debit/credit error from Odoo.

Reply:
I think this will now lead to an adjustment line here: https://github.com/OCA/edi/blob/60742e0cf64807a4981cdd13fe2e51f51235b4a8/account_invoice_import/components/edi_input_process_business_document_account_invoice.py#L291. Is that alright for you?